src/middlewared/middlewared/service/config_service.py

```diff
@@ -29,17 +35,62 @@ def __new__(cls, name, bases, attrs):
 
         namespace = klass._config.namespace.replace('.', '_')
         config_entry_key = f'{namespace}_entry'
-
-        if klass.ENTRY == NotImplementedError:
-            klass.ENTRY = Dict(config_entry_key, additional_attrs=True)
-
-        config_entry_key = klass.ENTRY.name
-
-        config_entry = copy.deepcopy(klass.ENTRY)
-        config_entry.register = True
-        klass.config = returns(config_entry)(klass.config)
+        config_model_name = f'{namespace.capitalize()}Config'
+
+        if klass._config.entry is not None or klass.ENTRY == NotImplementedError:
+            # FIXME: We do this temporarily to avoid breaking existing code
+            klass.ENTRY = None
+            if klass._config.entry is None:
+                klass._config.entry = create_model(
+                    f'{namespace.capitalize()}Entry',
```

Comment:
Autogenerated classes should absolutely not be the case with the versioned API approach we have now. Everything must be declared explicitly even if it is repetitive

Reply:
Okay so after hashing it out with @themylogin, decided approach is to not touch the `config.update` method but apply it to `config` method for it's result unless overridden in the child class.